ipv8/test/base.py

```diff
@@ -52,12 +52,12 @@ def setUp(self):
 
     def tearDown(self):
         try:
-            super(TestBase, self).tearDown()
             for node in self.nodes:
                 node.unload()
             internet.clear()
         finally:
             shutil.rmtree("temp", ignore_errors=True)
+        super(TestBase, self).tearDown()
```

Comment:
This should be in the `try`.

Reply:
Why?

Reply:
My original argument three days ago was that `rmtree` can take a while to complete, so the test class should not wait for it.

After some reflection time, we might as well leave this as is. Then we are at least sure one temp dir is gone before the next test case starts.